lib/gitsh/tab_completion/automaton_factory.rb

```diff
@@ -13,20 +13,22 @@ def initialize(env)
       end
 
       def build
-        load_config(File.join(env.config_directory, 'completions'))
+        start_state = Automaton::State.new('start')
+        config_paths.each do |path|
+          DSL.load(path, start_state, env)
+        end
         Automaton.new(start_state)
       end
 
       private
 
       attr_reader :env
 
-      def load_config(path)
-        DSL.load(path, start_state, env)
-      end
-
-      def start_state
-        @start_state ||= Automaton::State.new('start')
+      def config_paths
+        [
+          File.join(env.config_directory, 'completions'),
+          File.join(ENV['HOME'], '.gitsh_completions'),
```

Comment:
So there's an unlikely but possible chance that `$HOME` is unset -- for example, if I set `gitsh` as my login shell, `$HOME` may be unset.

That would mean that it reads the `.gitsh_completions` from `.`, I think. That feels a bit unpredictable -- the exploit there is that if I run gitsh from an untrusted directory, someone could drop a malformed gitsh_completions file in there.

OTOH, it might be handy -- you can get project-specific completions depending on where you start running gitsh. In which case `:cd` should also do this.

Either way, how about being more exact here:

```suggestion
          File.join(ENV.fetch('HOME', '/'), '.gitsh_completions'),
```

or

```suggestion
          File.join(ENV.fetch('HOME', '.'), '.gitsh_completions'),
```

Reply:
Good catch, thanks. I think falling back to `/.gitsh_completions` makes sense when `$HOME` isn't set—that would also be consistent with what would happen with the `.gitshrc` file.

Project-specific completions are an interesting idea, but I'm not sure I see a use case that would only make sense for a specific project.